libsolidity/ast/Types.cpp

```diff
@@ -1872,12 +1792,35 @@ bool ArrayType::isDynamicallyEncoded() const
 	return isDynamicallySized() || baseType()->isDynamicallyEncoded();
 }
 
-bigint ArrayType::storageSizeUpperBound() const
+bigint ArrayType::sizeUpperBound(DataLocation _loc) const
 {
 	if (isDynamicallySized())
 		return 1;
-	else
+
+	if (_loc == DataLocation::Storage)
 		return length() * baseType()->storageSizeUpperBound();
+
+	if (_loc == DataLocation::CallData)
+		return unlimitedStaticCalldataSize(true);
+
+	assert(_loc == DataLocation::Memory);
+	bigint size = bigint(length());
```

Comment:
How will the compiler handle memory types with sizes bigger than `bigint` range but still fitting in `u256`? Won't they get negative here? I know that for stuff bigger than `u256` the parser will say that it's not a valid integer but what happens for numbers in this gray area?

Reply:
> bigger than bigint range but still fitting in u256

Can't bigint handle all of `u256`?

Reply:
Oh, you're right. I confused `bigint` with `s256`. `bigint` is arbitrary precision so there should be no problems.